src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala

```diff
@@ -1353,4 +1354,21 @@ trait SilverTransforms extends SparkSessionWrapper {
 
   }
 
+  protected def buildWarehouseSpec(
+                                  bronze_warehouse_snap: PipelineTable,
+                                  isFirstRun: Boolean,
+                                  silver_warehouse_spec: PipelineTable
+                                )(df: DataFrame): DataFrame = {
+    val lastWarehouseSnapW = Window.partitionBy('organization_id, 'warehouse_id)
+      .orderBy('Pipeline_SnapTS.desc)
+
+    val bronzeWarehouseSnapLatest = bronze_warehouse_snap.asDF
+      .withColumn("rnk", rank().over(lastWarehouseSnapW))
```

Comment:
I see we do this in other modules as well, why do we use both rank as well as row_number instead of just row_number?

Reply:
as discussed we will leave this for now, until we get an answer/logic behind this ;)